sphinx/writers/latex.py

```diff
@@ -2294,6 +2298,9 @@ def visit_inline(self, node):
         elif classes in [['accelerator']]:
             self.body.append(r'\sphinxaccelerator{')
             self.context.append('}')
+        elif classes and self.in_code:
```

Comment:
I think it is better to use Transform module to propagation the inline nodes under literal node instead of remember its status.
```
class InlineCodeTransform(SphinxTransform):
    default_priority = 200

    def apply(self):
        for node in self.document.traverse(nodes.literal):
            for subnode in node:
                if isinstance(subnode, nodes.inline):
                    # mark inline nodes just under literal node as "code" to handle them in writer!
                    subnode['code'] = True
```

This makes writer simple. We have some transform modules for LaTeX writer in `sphinx.builders.latex.transforms`.

Reply:
I've update the PR with this code.